web/src/components/studies/StudyForm.tsx

```diff
@@ -2,74 +2,150 @@ import { useState, useEffect } from "react";
 
 export type StudyFormData = {
   title: string;
   description: string;
   owner: string;
   additionalStudyAdminUsernames: { value: string }[];
   dataControllerOrganisation: string;
-  cagReference: number;
+  cagReference: string | undefined | null;
   dataProtectionPrefix: string;
   dataProtectionDate: string;
   dataProtectionId: number;
-  dataProtectionNumber: string;
-  nhsEnglandReference: number;
-  irasId: string;
-  involvesUclSponsorship: boolean;
-  involvesCag: boolean;
-  involvesEthicsApproval: boolean;
-  involvesHraApproval: boolean;
-  requiresDbs: boolean;
-  isDataProtectionOfficeRegistered: boolean;
-  involvesThirdParty: boolean;
-  involvesExternalUsers: boolean;
-  involvesParticipantConsent: boolean;
-  involvesIndirectDataCollection: boolean;
-  involvesDataProcessingOutsideEea: boolean;
-  isNhsAssociated: boolean;
-  involvesNhsEngland: boolean;
-  involvesMnca: boolean;
-  requiresDspt: boolean;
+  dataProtectionNumber: string | undefined | null;
+  nhsEnglandReference: number | undefined | null;
+  irasId: string | undefined | null;
+  involvesUclSponsorship: boolean | null;
+  involvesCag: boolean | null;
+  involvesEthicsApproval: boolean | null;
+  involvesHraApproval: boolean | null;
+  requiresDbs: boolean | null;
+  isDataProtectionOfficeRegistered: boolean | null;
+  involvesThirdParty: boolean | null;
+  involvesExternalUsers: boolean | null;
+  involvesParticipantConsent: boolean | null;
+  involvesIndirectDataCollection: boolean | null;
+  involvesDataProcessingOutsideEea: boolean | null;
+  isNhsAssociated: boolean | null;
+  involvesNhsEngland: boolean | null;
+  involvesMnca: boolean | null;
+  requiresDspt: boolean | null;
 };
 
-type CreateStudyProps = {
+type StudyProps = {
   username: string;
-  setCreateStudyFormOpen: (name: boolean) => void;
-  handleStudySubmit: (data: StudyFormData) => Promise<void>;
-  submitError: string | null;
-  isSubmitting: boolean;
-  setSubmitError: (error: string | null) => void;
+  setStudyFormOpen: (name: boolean) => void;
+  fetchStudyData?: (id?: string) => void;
+  editingStudy?: Study | null;
 };
 
-// what is this?
-// A brief comment for context might be good here
+const convertStudyFormDataToApiRequest = (data: StudyFormData) => {
+  console.log("cagref value", data.cagReference);
+  const studyData: StudyRequest = {
+    title: data.title,
+    description: data.description ? data.description : undefined,
+    data_controller_organisation: data.dataControllerOrganisation.toLowerCase(),
+    additional_study_admin_usernames: data.additionalStudyAdminUsernames
+      .map((admin) => admin.value.trim())
+      .map((username) => `${username}${domainName}`),
+    involves_ucl_sponsorship: data.involvesUclSponsorship !== undefined ? data.involvesUclSponsorship : undefined,
+    involves_cag: data.involvesCag !== undefined ? data.involvesCag : undefined,
+    cag_reference: data.cagReference ? data.cagReference.toString() : undefined,
```

Comment:
If I submit a study with `involves_cag: true` a `cag_reference`, but then later edit my study to `involves_cag: false` the original `cag_reference` will remain. 

I'm not sure what the desired behaviour is here, but it's worth flagging now in case we want to remove any previously entered references.

I think this would also apply to the other conditional inputs too.

Reply:
https://github.com/ucl-arc-tre/portal/pull/316/commits/e5263bb73dea7d8b042a37aec9a61fcd0091e394

Reply:
> If I enter a reference number for the optional fields, but then set the parent field to no then the reference number remains in the database.

I think this is related to #385 

> It is impossible to set a form field back to unsure if any other answer was given previously, is this intentional?

no, but I think it's a side effect of not being able to update a value to be undefined